### PR TITLE
NIFI-10637 fixing flaky test in TestParseCEF

### PR DESCRIPTION
**Pull request by TheGreatRandall**
# Summary

[NIFI-10637](https://issues.apache.org/jira/browse/NIFI-10637)
Following the problem in the issue NIFI-10637. I dived deep to the class "ValidateLocale" where the validator has been determined valid or not.
After several printouts and checking log files, we found that the available locales is a deterministic list. 

```
final Locale[] availableLocales = Locale.getAvailableLocales();
```
However, the if condition in this class looks like this:
```
if (availableLocales[0].equals(testLocale)) {
                // Locale matches the "null" locale so it is treated as invalid
                return new ValidationResult.Builder().subject(subject).input(input).valid(false)
                        .explanation(input + " is not a valid locale format.").build();
            }
```
The purpose of this if statement is to determine if the testLocale is empty. However, as I said before, the availableLocales is a given list and the first element is not the zero. So I changed the code to this:

```
if ("".equals(testLocale.toString())) {
                // Locale matches the "null" locale so it is treated as invalid
                return new ValidationResult.Builder().subject(subject).input(input).valid(false)
                        .explanation(input + " is not a valid locale format.").build();
            }

```
By changing this, the if statement can correctly work and the bug has been fixed. It is not flaky anymore and can pass the NonDex's check.
# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the contribution @TheGreatRandall. This looks like a simple improvement. With the removal of `availableLocales[0]`, it looks like the line calling `Locale.getAvailableLocales()` could also be removed. Can you check that and make an update?

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseCEF.java [hunk 1]
```diff
@@ -396,7 +396,7 @@ public ValidationResult validate(String subject, String input, ValidationContext
             final Locale[] availableLocales = Locale.getAvailableLocales();
```
[the remark this answers is not in the document]
Comment (TheGreatRandall) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseCEF.java:396]: Thank you for your feedback! I'll update it very soon!

[the remark this answers is not in the document]
Comment (TheGreatRandall) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseCEF.java]: hi, I've removed the variable. Thank you!

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the updates @TheGreatRandall. Please review the following checkstyle warning:

```
Warning:  src/main/java/org/apache/nifi/processors/standard/ParseCEF.java:[396] (regexp) RegexpSinglelineJava: Line has trailing whitespace.
```

**Review by exceptionfactory — APPROVED** [review]
Removing the empty line cleared the Checkstyle validation, so this looks good. Thanks @TheGreatRandall! +1 merging

**Pull request closed without merging**